astropy/stats/circstats.py

```diff
@@ -134,6 +135,45 @@ def circmean(
     return _angle(data, 1.0, 0.0, axis, weights)
 
 
+def circmedian(data: Quantity | NDArray, axis: int | None = None) -> Quantity | NDArray:
+    """Computes the circular median angle of an array of circular data.
+
+    Parameters
+    ----------
+    data : ndarray or `~astropy.units.Quantity`
+        Array of circular (directional) data, which is assumed to be in
+        radians whenever ``data`` is `~numpy.ndarray`.
+    axis : int, optional
+        Axis along which circular medians are computed. The default is to compute
+        the median of the flattened array.
+
+    Returns
+    -------
+    ndarray or `~astropy.units.Quantity`
+        Circular median.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats import circmedian
+    >>> from astropy import units as u
+    >>> data = np.array([51, 67, 40, 109, 31, 405, 358])*u.deg
+    >>> circmedian(data)
+    <Quantity 45. deg>
+
+    References
+    ----------
+    .. [1] https://insidebigdata.com/2021/02/12/circular-statistics-in-python-an-intuitive-intro/
+    """
+    theta = np.arctan2(
+        np.median(np.sin(data), axis=axis),
```

Comment:
This feels not quite right, and indeed I could fairly easily find an example where it does not give the right answer:
```
In [11]: a = np.linspace(0, 135, 1001) << u.deg

In [12]: np.arctan2(np.median(np.sin(a)), np.median(np.cos(a))) << u.deg
Out[12]: <Quantity 65.27434311 deg>
```
(should be 67.5, as correctly found by replacing .median with .mean)

Note that ref. [1] also does something quite different.